src/vtkIECTransformLogic.h

```diff
@@ -157,7 +157,44 @@ class VTK_IEC_TRANSFORM_LOGIC_EXPORT vtkIECTransformLogic : public vtkObject
   {
     return this->IECTransforms;
   }
+  
+  /// Converts a vectorized index position to linear index position of a pixel in a DICOM image
+  /// @param i - the pixel column number 
+  /// @param j - the pixel row number
+  /// @param k - the image slice number 
+  /// @param nI - number of columns per image 
+  /// @param nJ - number of rows per image
+  /// @param nK - number of image slices 
+  /// \return The linearised pixel index as a single int
+  ///      NOTE: This algorithm uses row-major ordering to calculate indices
+  static inline  uint64_t VectorizedToLinearizedImageIndex(uint16_t i, uint16_t j , uint16_t k, uint16_t nI, uint16_t nJ, uint16_t nK)
+  {
+    if(i >= nI || j>= nJ || k >= nK)
+    {
+    	throw std::runtime_error("Indices out of range");
+    }
+    return static_cast<uint64_t>(k)*nI*nJ + static_cast<uint64_t>(j)*nI + static_cast<uint64_t>(i);
+  }
 
+  /// Converts a linear index position to a vectorized index position of a pixel in an DICOM image
+  /// @param linearizedindex - the linear index to be converted 
+  /// @param nI - number of columns per image 
+  /// @param nJ - number of rows per image
+  /// @param nK - number of image slices 
```

Comment:
```suggestion
  /// @param nI number of columns per image 
  /// @param nJ number of rows per image
  /// @param nK number of image slices 
```
Doxygen will format it for you, no need to add hyphens.